File: 08. Analisis Kompleksitas/B. Cek Bilangan Prima/b_cek_bilangan_prima.cpp
#include <iostream>
int main(){
  int N, num;
  bool isPrime;
  std::cin >> N;
  for (int i = 0; i < N; i++){
    std::cin >> num;
    isPrime = true;
    if (num == 1){
      isPrime = false;
    }
    else{
      for (int j = 2; j * j <= num; j++){
        if (num % j == 0){
          isPrime = false;
          break;
        }
      }
    }
    isPrime == true ? std::cout << "YA" << std::endl : std::cout << "BUKAN" << std::endl;
  }
  return 0;
}

File: 08. Analisis Kompleksitas/B. Cek Bilangan Prima/b_cek_bilangan_prima.pas
program b_cek_bilangan_prima;
var
  N, i, j : integer;
  num : longint;
  isPrime : boolean;
begin
  readln(N);
  for i := 1 to N do begin
    readln(num);
    isPrime := true;
    if (num = 1) then
      isPrime := false
    else
      for j := 2 to trunc(sqrt(num)) do begin
        if (num mod j = 0) then begin
          isPrime := false;
          break;
        end;
      end;
    if (isPrime) then
      writeln('YA')
    else
      writeln('BUKAN');
  end;
end.
